lib/cloudantBaseService.ts

```diff
@@ -115,6 +120,7 @@ export default abstract class CloudantBaseService extends BaseService {
    */
   public setServiceUrl(url: string): void {
     super.setServiceUrl(url);
+    CloudantBaseService.setAgent(this.baseOptions);
```

Comment:
Since this new version doesn't have to check the protocal on the URL (because it happens in `axios` anyway) we don't need to call this when the URL changes because new options are not passed at this time (and if options have been changed by the user in the mean time we should leave them alone).
```suggestion
```

Reply:
69652b5